From 49fd22a0661f94622710f465b525dd59d5699655 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Thu, 11 Oct 2018 14:27:34 +0300
Subject: [PATCH 0804/1051] misc: mvebu: move marvell spesific drivers to mvebu
 folder

This patch move SOC info driver to mvebu folder, as preperation to
add debugfs.

Change-Id: I2879f0972532fbeba62beef87c28cab9bd3520ad
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/60502
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Igal Liberman <igall@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06.
Drop the duplicate define of MVEBU_SOC_INFO.]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/misc/Kconfig                   | 8 +-------
 drivers/misc/Makefile                  | 2 +-
 drivers/misc/mvebu/Kconfig             | 8 ++++++++
 drivers/misc/mvebu/Makefile            | 1 +
 drivers/misc/{ => mvebu}/mv_soc_info.c | 0
 5 files changed, 11 insertions(+), 8 deletions(-)
 create mode 100644 drivers/misc/mvebu/Kconfig
 create mode 100644 drivers/misc/mvebu/Makefile
 rename drivers/misc/{ => mvebu}/mv_soc_info.c (100%)

diff --git a/drivers/misc/Kconfig b/drivers/misc/Kconfig
index 310163084d9c..f788974231fc 100644
--- a/drivers/misc/Kconfig
+++ b/drivers/misc/Kconfig
@@ -566,13 +566,7 @@ config MARVELL_OTX_BPHY_CTR
 	  The purpose of this is to handle some BPHY Interrupts in
 	  user space directly without kernel's intervention.
 
-config MVEBU_SOC_INFO
-	bool "Marvell Aramda-8K SoC information driver"
-	depends on ARCH_MVEBU
-	default y
-	help
-	  Marvell A8K SoC information driver
-
+source "drivers/misc/mvebu/Kconfig"
 source "drivers/misc/c2port/Kconfig"
 source "drivers/misc/eeprom/Kconfig"
 source "drivers/misc/cb710/Kconfig"
diff --git a/drivers/misc/Makefile b/drivers/misc/Makefile
index e62417a14536..5b391d34c6dd 100644
--- a/drivers/misc/Makefile
+++ b/drivers/misc/Makefile
@@ -60,4 +60,4 @@ obj-$(CONFIG_OCXL)		+= ocxl/
 obj-$(CONFIG_MISC_RTSX)		+= cardreader/
 obj-$(CONFIG_CAVIUM_TDM)	+= cavium-tdm.o
 obj-$(CONFIG_MARVELL_OTX_BPHY_CTR)	+= otx_bphy_ctr.o
-obj-$(CONFIG_MVEBU_SOC_INFO)	+= mv_soc_info.o
+obj-y				+= mvebu/
diff --git a/drivers/misc/mvebu/Kconfig b/drivers/misc/mvebu/Kconfig
new file mode 100644
index 000000000000..bed89bfe6248
--- /dev/null
+++ b/drivers/misc/mvebu/Kconfig
@@ -0,0 +1,8 @@
+# MISC MVEBU
+
+config MVEBU_SOC_INFO
+	bool "Marvell Aramda-8K SoC information driver"
+	depends on ARCH_MVEBU
+	default y
+	help
+	  Marvell A8K SoC information driver
diff --git a/drivers/misc/mvebu/Makefile b/drivers/misc/mvebu/Makefile
new file mode 100644
index 000000000000..5f4513ce35a8
--- /dev/null
+++ b/drivers/misc/mvebu/Makefile
@@ -0,0 +1 @@
+obj-$(CONFIG_MVEBU_SOC_INFO)	+= mv_soc_info.o
diff --git a/drivers/misc/mv_soc_info.c b/drivers/misc/mvebu/mv_soc_info.c
similarity index 100%
rename from drivers/misc/mv_soc_info.c
rename to drivers/misc/mvebu/mv_soc_info.c
-- 
2.17.1

